wire-ios-data-model/Source/Model/Message/ConversationMessage+Reaction.swift

```diff
@@ -18,111 +18,117 @@
 
 import Foundation
 
-@objc public enum MessageReaction: UInt16 {
-    // TODO: remove all cases and keep as string
-    case like
-    case thumbsUp
-    case thumbsDown
-    case slightlySmiling
-    case beamingFace
-    case frowningFace
-
-    public static func messageReaction(from unicodeValue: String) -> MessageReaction? {
-        switch unicodeValue {
-        case "❤️": return .like
-        case "👍": return .thumbsUp
-        case "🙂": return .slightlySmiling
-        case "😁": return .beamingFace
-        case "☹️": return .frowningFace
-        case "👎": return .thumbsDown
-        default: return nil
+extension ZMMessage {
+
+    @discardableResult
+    @objc public static func addReaction(
+        _ reaction: String,
+        toMessage message: ZMConversationMessage
+    ) -> ZMClientMessage? {
+
+        guard reaction != "" else {
+            return setReactions(
+                reactions: [],
+                toMessage: message
+            )
```

Comment:
Why remove all reactions if we try to set an empty reaction?

Reply:
If I remember correctly that was a fix for this [ticket](https://wearezeta.atlassian.net/jira/software/c/projects/WPB/boards/149?modal=detail&selectedIssue=WPB-3092&assignee=6170266af6da6a006a990919).

Reply:
Wouldn't it make more sense to return nil if we try to set an empty emoji?

Reply:
So instead of 

```swift
return setReactions(
                reactions: [],
                toMessage: message
            )
```

to ```return nil```

Reply:
Done!